script.R

```diff
@@ -15,8 +15,8 @@ clean_text <- function(text){
 
 build_translation_matrix <- function(){
   # the encoding rule is creating on the fly, as needed 
-  alphabet <- c('a', 'b', 'c', 'd','e','f','g','h','i','l','m','n','o','p','q','r','s','t','u','v','z')
-  code <- matrix(NA,ncol=6, nrow=21, byrow=F) %>% data.frame
+  alphabet <- c('a', 'b', 'c', 'd','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z')
+  code <- matrix(NA,ncol=6, nrow=26, byrow=F) %>% data.frame
```

Comment:
you need to make the code matrix composed by only 5 rows and change also the code_already_seen_for_this_letter function